Add --enable-fips flag for install command

frelon opened the pull request:
The --enable-fips flag will set the boot= and fips=1 kernel parameters
as well as running `fips-mode-setup --enable --no-bootcfg` during the
install procedure.

The running system shows the following output:

```
FIPS mode is enabled.
Initramfs fips module is enabled.
The current crypto policy (FIPS) is based on the FIPS policy.
```

Signed-off-by: Fredrik Lönnegren <fredrik.lonnegren@suse.com>

dirkmueller commented on pkg/bootloader/grub.go:
```diff
@@ -328,6 +328,15 @@ func (g *Grub) installKernelInitrd(rootPath, espDir, subfolder, snapshotID, kern
 	}
 
+	// Copy kernel .hmac in order to enable FIPS.
+	kernelHmac := filepath.Join(filepath.Dir(kernel), ".vmlinuz.hmac")
```
this should also check for `.Image.hmac` (for aarch64).

frelon replied:
Updated to use the kernel filename to look for the .hmac file.

frelon commented:
Linting fixed (I added `//nolint:gocyclo` to the upgrade method, since refactoring that method seems out of scope for this PR.)

Also updated the .hmac copying in bootloader to use the previously found kernel filename.

atanasdinov commented on pkg/deployment/deployment.go:
```diff
@@ -211,11 +211,16 @@ type FirmwareConfig struct {
 	Disks      []*Disk         `yaml:"disks"`
 	Firmware   *FirmwareConfig `yaml:"firmware"`
 	BootConfig *BootConfig     `yaml:"bootloader"`
+	Fips       *FipsConfig     `yaml:"fips"`
```
Do we expect more fields? If not, is a simple pointer to boolean enough?

atanasdinov commented on internal/cli/action/install.go:
```diff
@@ -136,7 +136,16 @@ func digestInstallSetup(s *sys.System, flags *cmd.InstallFlags) (*deployment.Dep
+			Enabled: true,
+		}
+
+		bootFlag := fmt.Sprintf("boot=LABEL=%s", deployment.EfiLabel)
```
Is this affected by FIPS enablement? I assumed we always need it.

atanasdinov commented on pkg/fips/fips.go:
```diff
@@ -0,0 +1,36 @@
+)
+
+func Enable(ctx context.Context, s *sys.System) error {
+	stdOut, err := s.Runner().RunContext(ctx, "/usr/bin/fips-mode-setup", "--enable", "--no-bootcfg")
```
I believe this is decommissioned [upstream](https://gitlab.com/redhat-crypto/fedora-crypto-policies/-/merge_requests/205#9ac97f30b729f8d1c92231a3f266fc15fbf8c8c6).

We are a couple of commits behind on [SLFO:Main](https://build.suse.de/projects/SUSE:SLFO:Main/packages/crypto-policies/files/crypto-policies.spec?expand=1) hence why the script is still present.

dirkmueller replied:
It also changes cryptopolicies. calling the command is still the recommended way for SL16, even if upstream decided to change things.
